### PR TITLE
[READY FOR REVIEW]Da botany update: Four new plants and plant traits + trait immutability

### PR DESCRIPTION
**Pull request by Krysonism**
## About The Pull Request

This pull request introduces 4 new plants as well as new traits, and allows some plant genes to be unable to extracted with the gene modder. 

**Galaxy Thistle** - contains a new liver medicine and has a trait that lets it spread itself in adjacent trays. Both  genes are extractable.
**Bungo Tree** - The fruit contains universal enzyme, but the pit contain a deadly cardiotoxin.
**Spaceman's Trumpet** - This flower contains a pigment that is intended as a pharmaceutical intermediate, for now you can grow it for its mild healing and lung restoring properties. reagent gene is unextractable.
**Jupiter Cup Mushroom** - This red mushroom loves pests and contains liquid electricity. Both genes unextractable. 

Plants now take more damage if the pest level is really high(8 or more) and plants are able to take pest damage will having high tox something that was not possible before for some reason.

## Why It's Good For The Game
Too often spacemen opt to put all the good genes into an easy to grow plant like wheat. By making some genes unextractable we encourage botanists to make choices of which plant to serve as the base for their super healing/death plants. This also opens up for the possibility of giving a strong plant a negative trait that makes the plant more challenging or interesting to grow, giving hydroponics gameplay a bit more depth.

Three of the four new plants also contain reagents that interact with the new organ damage system in some way, since that design space was largely unfilled.

## Changelog
:cl:
add: Four new plants!
add: Three new plant reagents!
tweak: Plants with high toxicity are now capable of taking pest damage.
balance: High pest levels are more damaging to plants.
imageadd: Added new plants sprites.
code: Coders can now make plant genes unextractable.
/:cl:
bungotree64
niceplant64
thistle
jupitercup64

### CODE REVIEW

**Comment by bigfatbananacyclops** [comment]
by adjacent, you mean connected or just close to eachother?

**Comment by Krysonism** [comment]
> 
> 
> by adjacent, you mean connected or just close to eachother?

Currently if the tray is right next to another tray or soil in a cardinal direction. I will probably change it to allow for diagonal spreading as well.

**Comment by Krysonism** [comment]
I will add an image of the new mushroom once i have eaten my dinner.

**Comment by Tlaltecuhtli** [comment]
turret plants when

**Comment by Krysonism** [comment]
> 
> 
> turret plants when

That would be sick. We could make one that throws it fruit at the nearest non-plant faction target when is ready to harvest. That might be easier to get merged.

File: code/__DEFINES/machines.dm [hunk 1]
```diff
@@ -106,3 +106,8 @@
 //cloning defines. These are flags.
 #define CLONING_SUCCESS (1<<0)
 #define CLONING_DELETE_RECORD (1<<1)
+
+//these are used to tell the DNA modifier if a plant gene cannot be extracted or modified.
+#define PLANT_GENE_IMMUTABLE	0
+#define PLANT_GENE_REMOVABLE	1
+#define PLANT_GENE_EXTRACTABLE	2
```
Comment (nemvar) [inline code/__DEFINES/machines.dm]: These can probably benefit from being flags.

Comment (Krysonism) [reply code/__DEFINES/machines.dm]: What is the advantage over using defines?

Comment (nemvar) [reply code/__DEFINES/machines.dm]: I meant that you could put flags on the genes that tell you, this can't be removed or this can't be extracted.

The benefit of flag is that you can apply all of them at once. For example it would be way easier to make a gene that can't be extracted or removed by just doing ``gene_flags = GENE_NOEXTRACT|GENE_NOREMOVE``

The benefit of doing that is that is easier to expand upon.

**Comment by MrDoomBringer** [comment]
Really cute sprites!

**Comment by Krysonism** [comment]
> 
> 
> Really cute sprites!

Thanks, it means at lot coming from you.

**Comment by shellspeed1** [comment]
So, quick question, will the Spaceman's Trumpet be wearable like other flowers and plants?

**Comment by optimumtact** [comment]
cool more grief tools for botany

**Comment by Ratvar** [comment]
Why is liquid electricity powercreep a good addon, what does it serve? I love other unique plants, but radcups 2.0 roundstart with downside of "you have to add traits TO this"? Why?

**Comment by CPTANT** [comment]
> 
> 
> cool more grief tools for botany

Literally all of these are beneficial except for the bungotoxin

**Comment by Tlaltecuhtli** [comment]
is the last plant from minecraft?

**Comment by Krysonism** [comment]
> 
> 
> Why is liquid electricity powercreep a good addon, what does it serve? Radcups 2.0 roundstart with downside of "you have to add traits TO this"? Why?
> Same question for roundstart "toxin that kills real fast, making you undefibbable and giving you confusion as only warning".
> Other plants look real nice and unique, together with traits and unextractability.

Both plants are mutations and thus wont be available roundstart and both plant are obvious toxin plants since they can't hide the toxins in death carrots.

Bungotoxin also does not stack well with the incredibly powerful toxins botany already has (coniine, entropic, bathsalts, zombie powder, amanitin, flurosulfuric).

Bungotoxin also requires it be mixed with something that causes dizziness in order to activate the confusion component and organ damage does not cause slowdown like damage from normal toxins.

**Comment by Krysonism** [comment]
> 
> 
> cool more grief tools for botany

I thought you wanted to focus development efforts on departmental gameplay, oranges.

If people grief using bungotoxin to explode peoples hearts i expect them to get banned.

**Comment by CPTANT** [comment]
I don't really see how powerful the bungotoxin is from the numbers.

What happens if you inject 15u into someone?

**Comment by Ratvar** [comment]
So bungotoxin has *no* tell that it's killing you? Guh, amantoxin 2.0

**Comment by Krysonism** [comment]
> 
> 
> So bungotoxin has _no_ tell that it's killing you? Guh, amantoxin 2.0

I think i will add a tell and bump the damage up to 3.

Considering that bathsalts do 4 brain, makes you step in a  random direction and makes you hallucinate, with a super deadly addiction after if you manage to survive i think that would be fair.

**Comment by Farquaar** [comment]
The sprites look nice.

File: code/__DEFINES/machines.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/__DEFINES/machines.dm]: what is an immutable gene?

Removable/Extractable overlap so those wouldn't be good for flags since to extract a gene you need to remove it from the seed, ruining it in the process.

I guess you could just refluff it if that's the only issue but 🤷‍♂

**Comment by ExcessiveUseOfCobblestone** [comment]
did u make these kryo? frikken rad sprites

**Comment by Fikou** [comment]
snowflake unextractable genes bad

**Comment by Krysonism** [comment]
>did u make these krys? frikken rad sprites

Thank you.

Yes, I was always more of a sprite guy than a coderman.

>snowflake unextractable genes bad
 Why?

Let me expand upon why I think this is a good change. The current "meta" is to turn botany into a monoculture of GMO wheat or another easy to grow plant. Any attempt to make a new plant with new interesting growth requirements is just going to be sidestepped. We can also prevent powercreep by locking two traits that are problematic together to different plants.

Encouraging botanists to grow a more diverse set of plants also makes oversight easier as the AI or HoP can inspect the botany and see if they are growing anything shady and tell sec to be vigilant.

The idea is not to get rid of GMO plants but to make botanists make meaningful choices as to which plant to use as a "platform".

One interesting application I am looking into it coding is atmos requirement genes for some high end plants. You could have plant that needs high air humidity to grow or maybe some CO2, wouldn't it be cool if you could make climate controlled growing rooms, maybe making half of botany Into a corpse farm for growing miasma loving plant or coplanting them with corpseflowers?

**Comment by ExcessiveUseOfCobblestone** [comment]
People still use wheat in 2559? yikes...

File: code/modules/hydroponics/grown/cocoa_vanilla.dm [hunk 2]
```diff
@@ -50,3 +50,47 @@
 	foodtype = FRUIT
 	tastes = list("vanilla" = 1)
 	distill_reagent = /datum/reagent/consumable/vanilla //Takes longer, but you can get even more vanilla from it.
+
+/obj/item/seeds/cocoapod/bungotree
+	name = "pack of bungo tree seeds"
+	desc = "These seeds grow into bungo trees. They look like they might contain a horrible cardiotoxin."
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/cocoa_vanilla.dm]: how does a plant look like it might contain a horrible cardiotoxin?

File: code/modules/hydroponics/grown/cocoa_vanilla.dm [hunk 3]
```diff
@@ -50,3 +50,47 @@
...
+	reagents_add = list(/datum/reagent/consumable/enzyme = 0.15, /datum/reagent/consumable/nutriment = 0.1)
+	growthstages = 4
+	growing_icon = 'icons/obj/hydroponics/growing_fruits.dmi'
+	icon_grow = "bungotree-grow"
+	icon_dead = "bungotree-dead"
+
+/obj/item/reagent_containers/food/snacks/grown/bungofruit
+	seed = /obj/item/seeds/cocoapod/bungotree
+	name = "bungo fruit"
+	desc = "A strange fruit, tough leathery skin protects Its juicy flesh and large poisonous seed."
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/cocoa_vanilla.dm]: uncapitalize "Its"

File: code/modules/hydroponics/grown/cocoa_vanilla.dm [hunk 4]
```diff
@@ -50,3 +50,47 @@
...
+	filling_color = "#E8C22F"
+	foodtype = FRUIT
+	tastes = list("bungo" = 2, "tropical fruitiness" = 1)
+	distill_reagent = null
+
+/obj/item/reagent_containers/food/snacks/grown/bungopit
+	seed = /obj/item/seeds/cocoapod/bungotree
+	name = "bungo pit"
+	icon_state = "bungopit"
+	desc = "A large seed, It is said to be potent enough to be able to stop a mans heart."
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/cocoa_vanilla.dm]: uncapitalize "It"

File: code/modules/hydroponics/grown/flowers.dm [hunk 5]
```diff
@@ -214,3 +214,35 @@
 	if(!user.gloves)
 		to_chat(user, "<span class='danger'>The [name] burns your bare hand!</span>")
 		user.adjustFireLoss(rand(1, 5))
+
+/obj/item/seeds/poppy/lily/trumpet
+	name = "pack of spaceman's trumpet seeds"
+	desc = "A plant sculped by extensive genetic engineering. The spaceman's trumpet is said to bear no resemblance to its wild ancestors. Inside the NT central AgriSci department it is better known as NTPW-0372."
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/flowers.dm]: "the NT central AgriSci department" looks a bit weird

File: code/modules/hydroponics/grown/flowers.dm [hunk 6]
```diff
@@ -214,3 +214,35 @@
...
+	reagents_add = list(/datum/reagent/consumable/nutriment = 0.05)
+
+/obj/item/seeds/poppy/lily/trumpet/Initialize()
+	..()
+	set_mutability(/datum/plant_gene/reagent/polypyr, PLANT_GENE_REMOVABLE)
+
+/obj/item/reagent_containers/food/snacks/grown/trumpet
+	seed = /obj/item/seeds/poppy/lily/trumpet
+	name = "spaceman's trumpet"
+	desc = "A vivid flower that smells faintly of.freshly cut grass.Touching the flower seems to stain the skin some time after contact, yet most other surfaces seem to be unaffected by this phenomenon."
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/flowers.dm]: remove the period after the "of" and add a space after the period after "grass"

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 7]
```diff
@@ -899,3 +899,18 @@
...
+	description = "A horrible cardiotoxin that protects the humble bungo pit."
+	silent_toxin = TRUE
+	color = "#EBFF8E"
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
+	toxpwr = 0
+	taste_description = "tannin"
+
+/datum/reagent/toxin/bungotoxin/on_mob_life(mob/living/carbon/M)
+	M.adjustOrganLoss(ORGAN_SLOT_HEART, 3)
+	M.confused = M.dizziness //add a tertiary effect here if this is isn't an effective poison.
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: uh, why is this line here? is this supposed to be combo'd with some dizzying chem or something? If you're not dizzy, won't this near-instantly "cure" any confusion that you might have?

File: code/modules/hydroponics/grown/cocoa_vanilla.dm [hunk 8]
```diff
@@ -50,3 +50,47 @@
...
+	throw_speed = 3
+	throw_range = 7
+	foodtype = TOXIC
+	tastes = list("acrid bitterness" = 1)
+
+/obj/item/reagent_containers/food/snacks/grown/bungopit/Initialize()
+	. =..()
+	reagents.clear_reagents()
+	reagents.add_reagent(/datum/reagent/toxin/bungotoxin, seed.potency * 0.15) //replace with new toxin
+	reagents.add_reagent(/datum/reagent/consumable/nutriment, seed.potency * 0.04)
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/cocoa_vanilla.dm]: do "trash" plants copy their parent plant's traits? if so, prepare for epic hypodermic prickles+slipping+horrifically lethal cardiotoxin memes from traitor botanists (this might not be a bad thing, mind)

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm (hunk 7 above)
Comment (Krysonism) [reply code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: Yes

File: code/modules/hydroponics/grown/cocoa_vanilla.dm (hunk 2 above)
Comment (Krysonism) [reply code/modules/hydroponics/grown/cocoa_vanilla.dm]: I just thought it was funny, it is based on the cocoa tree description. I will change it.

File: code/modules/hydroponics/grown/cocoa_vanilla.dm (hunk 8 above)
Comment (Krysonism) [reply code/modules/hydroponics/grown/cocoa_vanilla.dm]: Yes, and they actually transfer way too much toxin on slip(all 15u at 100 potency). So thanks for reminding me to test that.

I think 3-4u is more appropriate

**Comment by HomocidlePancakes** [comment]
Netherwarts. Nice.

**Comment by ExcessiveUseOfCobblestone** [comment]
bro if u are making actual grammar fixes use the suggestions feature.

**Comment by Krysonism** [comment]
I have been busy at work this week so i have not been able to code.

I have the made purple dye in the spaceman's trumpet able to dye your hair and beard purple. 

Disguise as a snowflake without getting George Melon's beard.

The bungo fruit also has less enzyme as it was overtaking the flavours of the nutriment.

Things I'd like to address before i consider this ready:

Replace the defines with flags
Make the Invasive spreading spread diagonally as well.
Grammar and spelling
Fix the bungo pit injecting all its reagents at once.

**Comment by ExcessiveUseOfCobblestone** [comment]
how do you get the trumpet seed? Not sure i'm a fan of the chem depending on the ease of obtaining

Also trumpet should be placed with the poppy subtypes instead at the bottom of the file.

**Comment by Krysonism** [comment]
>how do you get the trumpet seed?

Mutate poppy into lilly, then mutate lilly into trumpet.(as hard to get as coniine)

**Comment by ExcessiveUseOfCobblestone** [comment]
am I missing something?

https://github.com/tgstation/tgstation/pull/45840/files#diff-a09de7a0ff088df4c3dbc49a3e9f28feR39

**Comment by Krysonism** [comment]
> 
> 
> am I missing something?
> 
> https://github.com/tgstation/tgstation/pull/45840/files#diff-a09de7a0ff088df4c3dbc49a3e9f28feR39

No, i have missed adding it as a mutation. I will make sure, all the mutate lists  are okay.

Cocoa pod  -> Bungo tree
Chantarelle -> Jupiter cups
Poppy->Lilly->Trumpet
Star thistle->Galaxy thistle

File: code/modules/hydroponics/grown/cocoa_vanilla.dm (hunk 8 above)
Comment (Krysonism) [reply code/modules/hydroponics/grown/cocoa_vanilla.dm]: I have since learned that vanilla 100 potency death nettles also kills you in one hit.

I will make sure to read the hypodermic prickles code to make sure nothing is acting up though.

**Comment by Krysonism** [comment]
Should i change the name mutability_flags to something a bit more generic?

I think what i have left before this is ready is to update the documentation, make the thistle spread diagonally and read up on hypodermic prickles to make sure it is working properly with the bungo pit.

**Comment by Krysonism** [comment]
I think this is pretty much ready for review

File: code/modules/hydroponics/plant_genes.dm [hunk 9]
```diff
@@ -1,13 +1,13 @@
...
 /datum/plant_gene/proc/get_name() // Used for manipulator display and gene disk name.
 	var/formatted_name
-	if(mutability == PLANT_GENE_IMMUTABLE)
-		formatted_name += "Immutable "
-	else if(mutability == PLANT_GENE_REMOVABLE)
-		formatted_name += "Excisable "
+//	if(mutability == PLANT_GENE_IMMUTABLE)
+//		formatted_name += "Immutable "
+//	else if(mutability == PLANT_GENE_REMOVABLE)
+//		formatted_name += "Excisable "
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/plant_genes.dm]: commented code

File: code/modules/hydroponics/plant_genes.dm [hunk 10]
```diff
@@ -109,10 +109,10 @@
...
 /datum/plant_gene/reagent/get_name()
 	var/formatted_name
-	if(mutability == PLANT_GENE_IMMUTABLE)
-		formatted_name += "Immutable "
-	else if(mutability == PLANT_GENE_REMOVABLE)
-		formatted_name += "Excisable "
+//	if(mutability_flags & PLANT_GENE_IMMUTABLE)
+//		formatted_name += "Immutable "
+//	else if(mutability_flags & PLANT_GENE_REMOVABLE)
+//		formatted_name += "Excisable "
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/plant_genes.dm]: commented code

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

**Comment by Timonkeyn** [comment]
make the unextractable genes be extractable with Tier 5 or 6(admin only)

File: code/modules/hydroponics/plant_genes.dm (hunk 10 above)
Comment (Krysonism) [reply code/modules/hydroponics/plant_genes.dm]: I think I fixed that 4 days ago, or am I going insane?

**Comment by ExcessiveUseOfCobblestone** [comment]
@Krysonism it was showing me not the most recent commit for some reason, cringe.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm ok with C3 for now until I see it getting abused, then it's going through the grinder.
